lib/web/apiserver.go

```diff
@@ -2148,15 +2149,26 @@ func clientMetaFromReq(r *http.Request) *auth.ForwardedClientMetadata {
 	}
 }
 
-// deleteWebSession is called to sign out user
+// deleteWebSession is called to sign out user from web, app and SAML IdP session.
 //
 // DELETE /v1/webapi/sessions/:sid
 //
 // Response:
 //
 // {"message": "ok"}
 func (h *Handler) deleteWebSession(w http.ResponseWriter, r *http.Request, _ httprouter.Params, ctx *SessionContext) (interface{}, error) {
-	err := h.logout(r.Context(), w, ctx)
+	// TODO(sshah): we can skip this step below once we have a mechanism to update websession
+	// with SAML details.
+	samlSessionCookie, err := r.Cookie(samlidp.SAMLSessionCookieName)
+	if err != nil {
+		h.log.Debug("SAML IdP session not found for deletion for user: %s", ctx.GetUser())
```

Comment:
Use structured logging (also a "Debug" statement with format directive is an error).

```suggestion
		h.log.
			WithError(err).
			WithField("user", ctx.GetUser()).
			Debug("SAML IdP session not found for deletion for user")
```